### PR TITLE
httptransport: add robots.txt

### PR DESCRIPTION
**Pull request by hdonnay**
This adds an unauthenticated "/robots.txt" endpoint on the API server.

### CODE REVIEW

**Comment by hdonnay** [comment]
Current dependencies on/for this PR:
* `main`
  * **PR #1947** <a href="https://app.graphite.dev/github/pr/quay/clair/1947?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/quay/clair/1947?utm_source=stack-comment).

File: httptransport/server.go [hunk 1]
```diff
@@ -112,7 +112,12 @@ func New(ctx context.Context, conf *config.Config, indexer indexer.Service, matc
 				Msg("received error configuring auth middleware")
 			return nil, err
 		}
-		ret = h
+		final := http.NewServeMux()
+		final.Handle("/robots.txt", robotsHandler)
+		final.Handle("/", h)
+		ret = final
+	} else {
+		mux.Handle("/robots.txt", robotsHandler)
```
Comment (crozzy) [inline httptransport/server.go]: nit: Would this be more readable as `ret.Handle("/robots.txt", robotsHandler)`

**Review by crozzy — DISMISSED** [review]

**Comment by hdonnay** [comment]
@crozzy would like to get this in for the next release

**Review by crozzy — APPROVED** [review]
LGTM good idea using the commit date for the `Last-Modified` time

**Comment by hdonnay** [comment]
On second thought, not backporting this -- there's additional changes in the `httptransport` package that make this annoying.

**Pull request merged**